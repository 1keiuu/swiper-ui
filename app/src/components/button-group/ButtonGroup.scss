.button__group {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 375px;
  height: 80px;
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
  margin: 0 auto;
}
