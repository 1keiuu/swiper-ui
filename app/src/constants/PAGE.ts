export const PAGE_STATUS: {
  RELOADING: { name: PageStatus };
  EMPTY: { name: PageStatus };
} = {
  RELOADING: { name: "reloading" },
  EMPTY: { name: "empty" },
};
