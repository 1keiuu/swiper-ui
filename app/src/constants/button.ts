export const BUTTON_TYPES = {
  NOPE: { name: "nope" },
  PROFILE: { name: "profile" },
  LIKE: { name: "like" },
};
export const BUTTON_TYPE_OPTIONS: string[] = [
  BUTTON_TYPES.NOPE.name,
  BUTTON_TYPES.PROFILE.name,
  BUTTON_TYPES.LIKE.name,
];
export const STATUS_BUTTON_TYPES: CardStatus[] = ["nope", "like"];
