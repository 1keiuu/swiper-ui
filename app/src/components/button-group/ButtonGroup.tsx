import React, { useEffect, useState } from "react";
import "./ButtonGroup.scss";
import Button from "../button/Button";
import { BUTTON_TYPE_OPTIONS } from "../../constants/button";
type ButtonGroupProps = { isEmpty: boolean };
const ButtonGroup: React.FC<ButtonGroupProps> = (props) => {
  return (
    <div className="button__group">
      {BUTTON_TYPE_OPTIONS.map((type: ButtonTypeOption, i: number) => {
        return (
          <Button
            buttonType={type}
            key={`button${i}`}
            isEmpty={props.isEmpty}
          />
        );
      })}
    </div>
  );
};

export default ButtonGroup;
