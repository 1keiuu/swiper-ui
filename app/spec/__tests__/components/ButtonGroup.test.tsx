import React from "react";
import { cleanup, render } from "@testing-library/react";

import ButtonGroup from "../../../src/components/button-group/ButtonGroup";
import { BUTTON_TYPE_OPTIONS } from "../../../src/constants/BUTTON";

// HACK: constantのテストをここに書くべきか。個別で作るか。
describe("constant", () => {
  test("'BUTTON_TYPE_OPTIONS' exists", () => {
    expect(BUTTON_TYPE_OPTIONS.length).toBeGreaterThan(1);
  });
});

describe("ButtonGroup", () => {
  // HACK: ここモックデータにすべきか、constantからimportした物にすべきか
  const BUTTON_TYPE_OPTIONS: ButtonTypeOption[] = ["nope", "profile", "like"];
  afterEach(() => {
    cleanup();
  });
  test("render as before.", () => {
    const { asFragment } = render(<ButtonGroup isEmpty={false}></ButtonGroup>);
    expect(asFragment()).toMatchSnapshot();
  });
  test("contains children regardless of isEmpty state.", () => {
    const isEmptyButtonGroup = render(
      <ButtonGroup isEmpty={true}></ButtonGroup>
    );
    const isNotEmptyButtonGroup = render(
      <ButtonGroup isEmpty={false}></ButtonGroup>
    );
    expect(isEmptyButtonGroup.container.firstChild.childNodes.length).toBe(
      BUTTON_TYPE_OPTIONS.length
    );
    expect(isNotEmptyButtonGroup.container.firstChild.childNodes.length).toBe(
      BUTTON_TYPE_OPTIONS.length
    );
  });
  test("contains children.IN Addition, children's length changes depending on 'BUTTON_TYPE_OPTIONS'.", () => {
    // TODO: レンダリングされるボタンの数がBUTTON_TYPE_OPTIONSの数によって変化するかを確認したい。
    // 難点としては、BUTTON_TYPE_OPTIONSの数を変更するのが難しい。fsなど使って無理やり書き換えるか、そもそもの実装を変える?
  });
});
