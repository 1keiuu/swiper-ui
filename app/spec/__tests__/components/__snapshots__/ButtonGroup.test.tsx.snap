// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Component render as before. 1`] = `
<DocumentFragment>
  <div
    class="button__group"
  >
    <button
      class="button --nope"
    >
      <img
        alt="button icon"
        src="test-file-stub"
      />
    </button>
    <button
      class="button --profile"
    >
      <img
        alt="button icon"
        src="test-file-stub"
      />
    </button>
    <button
      class="button --like"
    >
      <img
        alt="button icon"
        src="test-file-stub"
      />
    </button>
  </div>
</DocumentFragment>
`;
